lua/tests/infoview_lsp_spec.lua

```diff
@@ -0,0 +1,62 @@
+local infoview = require('lean.infoview')
+local helpers = require('tests.helpers')
+
+local function get_info_lines(infoview_info)
+  return table.concat(vim.api.nvim_buf_get_lines(infoview_info.bufnr, 0, -1, true), "\n")
+end
+
+local function infoview_lsp_test(pos, expected)
+    local infoview_info = infoview.open()
+    vim.api.nvim_win_set_cursor(0, pos)
+    infoview.update(infoview_info.bufnr)
+    local result, _ = vim.wait(5000, function()
+      for _, string in pairs(expected) do
+        local curr = get_info_lines(infoview_info)
+        if not curr:find(string) then return false end
+      end
+      return true
+    end)
+    assert.message( "expected: " .. vim.inspect(expected) ..  ", actual: " .. get_info_lines(infoview_info)
+    ).is_true(result)
+end
+
+describe('infoview', function()
+  helpers.setup {
+    infoview = { enable = true },
+    lsp = { enable = true },
+    lsp3 = { enable = true },
+  }
+
+  it('lean 3', function()
+    vim.api.nvim_command("edit lua/tests/fixtures/example-lean3-project/test.lean")
```

Comment:
Maybe we should be using [`before_each`](https://github.com/nvim-lua/plenary.nvim/blob/master/TESTS_README.md#a-simple-test)?